tests/codegen/align-fn.rs

```diff
@@ -25,27 +25,60 @@ impl A {
 trait T: Sized {
     fn trait_fn() {}
 
-    // CHECK: align 32
-    #[align(32)]
     fn trait_method(self) {}
+
+    #[align(8)]
+    fn trait_method_inherit_low(self);
+
+    #[align(32)]
+    fn trait_method_inherit_high(self);
+
+    // CHECK-LABEL: trait_method_inherit_default
+    // CHECK-SAME: align 32
+    #[align(32)]
+    #[no_mangle]
```

Comment:
`no_mangle` on a trait declaration item is an FCW

Reply:
(Also, should probably be `unsafe(no_mangle)`)

Reply:
That's what the 

```rust
pub struct InstantiateDefaultMethods;

impl T for InstantiateDefaultMethods {
    fn trait_method_inherit_low(self) {}
    fn trait_method_inherit_high(self) {}
}
```

bit is for.

Reply:
We should really have a `#[rustc_force_codegen]` or something.

Reply:
I'm also not sure how well it would work here, but it would be nicer than `#[no_mangle]`, which is really trying to say an entirely different thing, and `-Clink-dead-code`.

Reply:
Yes, turns out that does work, but because the name is unmangled, rust inserts an extra comment that also contains the function name, so you have to take that into account when writing the `CHECK`s.

Reply:
> That's what the […] bit is for.

Imagine if the default body referenced `Self`, and you had two instantiations for different types. Then we would have two `no_mangle` symbols with the same name.

Reply:
> I just took some messing around to find a solution that doesn't need `no_mangle`.

Sorry, what solution?